src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/util/OcrFormValidationHelper.java

```diff
@@ -0,0 +1,62 @@
+package uk.gov.hmcts.reform.bulkscanccdeventhandler.util;
+
+import org.apache.commons.validator.routines.EmailValidator;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.model.OcrDataField;
+
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.toList;
+import static org.apache.commons.lang.StringUtils.isBlank;
+
+public final class OcrFormValidationHelper {
+
+    private OcrFormValidationHelper() {
+        // util class
+    }
+
+    public static List<String> getOcrFieldNames(List<OcrDataField> ocrData) {
+        return ocrData
+            .stream()
+            .map(field -> field.key)
+            .collect(toList());
+    }
+
+    public static List<String> findBlankFields(List<String> fieldNames, List<OcrDataField> ocrData) {
+        return fieldNames
+            .stream()
+            .filter(field -> isBlank(OcrFormValidationHelper.findOcrFormFieldValue(field, ocrData)))
+            .collect(toList());
+    }
+
+    public static List<String> findMissingFields(List<String> optionalFields, List<String> ocrInputFields) {
```

Comment:
We could rename parameters to something more general, e.g. `expectedFields` and `presentFields` or sth. MY recommendation would be to use `Sets.difference` instead, but up to you.

Reply:
I have to leave it as it is now. I've renamed the parameters as `expectedFields` and `inputFields`.
This method is to check if `inputFields` contains all `expectedFields` and return the missing `expectedFields`.  `Sets.difference` is returning more fields as `inputFields` includes other fields as well